web/apps/labelstudio/src/pages/Settings/StorageSettings/StorageSummary.jsx

```diff
@@ -37,34 +38,46 @@ export const StorageSummary = ({ target, storage, className, storageTypes = [] }
       `storage ${storage.id} in project ${storage.project} and job ${storage.last_sync_job}:\n\n` +
       `${getLastTraceback(storage.traceback)}\n\n` +
       `meta = ${JSON.stringify(storage.meta)}\n`;
-    const targetType = target === "export" ? "Target" : "Source";
+
+    // Component to handle copy functionality within the modal
+    const CopyButton = () => {
+      const [copyText, copied] = useCopyText(msg);
+
+      return (
+        <Button
+          icon={<IconFileCopy />}
+          onClick={copyText}
+          disabled={copied}
+          style={{ minWidth: "100px", justifyContent: "center" }}
+        >
+          {copied ? "Copied!" : "Copy"}
+        </Button>
+      );
+    };
```

Comment:
Components should always be defined at module level, not within another Component. Move this up and out, and use the component props to take the `msg`.